### ISSUE TITLE
Tests fail with click 8.2.0

### ISSUE THREAD

**Issue by s3v-** [sqlfluff/sqlfluff#6676]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Hi,

click 8.2.0 entered in Debian [1] and tests start failing because some [changes](https://github.com/pallets/click/blob/main/CHANGES.rst) in new click, in particular:

- "mix_stderr" param was removed from CliRunner() initializer
- _stdout_ and _stderr_ streams have been decoupled, so _result.output_ is empty in failing test and assertion should be against _result.stderr_
- the return code is 2 instead of 0 if a command is executed on the command line without parameters

**click < 8.2.0**

```shell
$ sqlfluff > /dev/null
$ echo $?
0

$ sqlfluff 2> /dev/null
Usage: sqlfluff [OPTIONS] COMMAND [ARGS]...

  SQLFluff is a modular SQL linter for humans.

Options:
  --version   Show the version and exit.
  -h, --help  Show this message and exit.

Commands:
 [...]
 
$ echo $?
0
```
**click >= 8.2.0**

```shell
$ sqlfluff > /dev/null
Usage: sqlfluff [OPTIONS] COMMAND [ARGS]...

  SQLFluff is a modular SQL linter for humans.

Options:
  --version   Show the version and exit.
  -h, --help  Show this message and exit.

Commands:
 [...]
 
$ echo $?
2

$ sqlfluff 2> /dev/null
$ echo $?
2
```

Failure log is visible here [2]

System info:
- Debian Sid
- Python 3.13
- click 8.2.0
- sqlfluff 3.3.1

Kind Regards.

[1] https://tracker.debian.org/news/1618122/accepted-python-click-820-1-source-into-unstable/
[2] https://ci.debian.net/packages/s/sqlfluff/testing/amd64/58037763/

### Expected Behaviour

Test to pass with click 8.2.0

### Observed Behaviour

Tests fail as spotted in the aforementioned log 

### How to reproduce

Simply run pytest against sqlfluff tests on a system where click 8.2.0 is installed.

### Dialect

None

### Version

3.3.1 sources, the package is not installed on the system

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
